### PR TITLE
TESTSUITE: Extract cmake logs in testresults.

### PR DESCRIPTION
**Pull request by SaillantNicolas**
## Summary of Changes

This PR moves and adds some CMake logs to the test results.
The changes of this Pull Request can be viewed [here](https://cgal.geometryfactory.com/~nsaillant/CmakeLogsTestsuite/TestReport_cgal_tester_Fedora)

## Release Management

* Issue(s) solved (if any): #8165

### CODE REVIEW

File: Testsuite/test/post_process_ctest_results.py [hunk 1]
```diff
@@ -3,12 +3,13 @@
 import re
 import os
 
-report_file=sys.argv[1]
-report_name=sys.argv[2]
-global_report_name=sys.argv[3]
-rx=re.compile('(.*Configuring (examples|demo|test)*( in )*(test\/|examples\/|demo\/)*)((?!done)\w+)')
-rx_demo=re.compile('.*in demo\/')
-rx_examples=re.compile('.*in examples\/')
+input_report_file=sys.argv[1]
```
Comment (lrineau) [inline Testsuite/test/post_process_ctest_results.py:6]: Better use `_name` as a suffix to `input_report_file`. Reserve `input_report_file` for the actual file descriptor once the file is open.

**Review by lrineau — COMMENTED** [review]
It seems `import io` is not used.

Actually, the linter pylint reports a lot of warning about this code.

**Comment by lrineau** [comment]
> This PR moves and adds some CMake logs to the test results. The changes of this Pull Request can be viewed [here](https://cgal.geometryfactory.com/~nsaillant/CmakeLogsTestsuite/TestReport_cgal_tester_Fedora)

@afabri @MaelRL @sloriot What do you think about the improvements? See here for an example: https://cgal.geometryfactory.com/~nsaillant/CmakeLogsTestsuite/TestReport_cgal_tester_Fedora

File: Testsuite/test/post_process_ctest_results.py [hunk 2]
```diff
@@ -1,75 +1,126 @@
...
-global_report_name=sys.argv[3]
-rx=re.compile('(.*Configuring (examples|demo|test)*( in )*(test\/|examples\/|demo\/)*)((?!done)\w+)')
-rx_demo=re.compile('.*in demo\/')
-rx_examples=re.compile('.*in examples\/')
+input_report_file_name=sys.argv[1]
+report_file_name=sys.argv[2]
+global_report_file_name=sys.argv[3]
+config_regex=re.compile('(.*Configuring (examples|demo|test)*( in )*(test\/|examples\/|demo\/)*)((?!done)\w+)')
+demo_regex=re.compile('.*in demo\/')
+examples_regex=re.compile('.*in examples\/')
```
Comment (lrineau) [inline Testsuite/test/post_process_ctest_results.py:10]: Please try with raw strings:
```suggestion
config_regex=re.compile(r'(.*Configuring (examples|demo|test)*( in )*(test/|examples/|demo/)*)((?!done)\w+)')
demo_regex=re.compile(r'.*in demo/')
examples_regex=re.compile(r'.*in examples/')

```

**Comment by afabri** [comment]
looks good to me.

**Comment by lrineau** [comment]
That is still wrong:
https://cgal.geometryfactory.com/CGAL/testsuite/results-6.0-Ic-261.shtml#Algebraic_kernel_d_Examples

**Comment by lrineau** [comment]
And Linux dockers all give an error:
```
Traceback (most recent call last):
  File "/mnt/testsuite/test/post_process_ctest_results.py", line 56, in <module>
    for myline in input_report_file:
ValueError: I/O operation on closed file.
```

See https://cgal.geometryfactory.com/CGAL/testsuite/results-6.0-Ic-261.shtml

**Comment by sloriot** [comment]
There are some strange lines in https://cgal.geometryfactory.com/CGAL/testsuite/results-6.0-Ic-316.shtml.

image

**Comment by lrineau** [comment]
@sloriot Ready to be rested.

**Comment by lrineau** [comment]
> @sloriot Ready to be re-tested.

Again

**Review by lrineau — APPROVED** [review]

**Review by lrineau — APPROVED** [review]
Looks good, now.

With Nicolas we have found and solved the issue on `magritte`.

It was:

```cmake
set(WITH_tests ON CACHE BOOL "")
```

in the "initial cache" for the platform.

**Pull request merged**